Remove logger from IContainerContext (#9147)
microsoft/FluidFramework: BREAKING.md

- Provide guidance on how the change should be consumed if applicable, such as by specifying replacement APIs.
- Consider providing code examples as part of guidance for non-trivial changes.

## 0.58 Breaking changes
- [Remove raiseContainerWarning property](#Remove-raiseContainerWarning-property)

> - [Remove logger property from IContainerContext](#Remove-logger-property-from-IContainerContext)

Reviewer: I meant to put this comment on the other list item ^^

> - [Remove logger property from IContainerContext](#Remove-logger-property-from-IContainerContext)

Reviewer 2: @sdeshpande3 we chatted with Helio around this. This should go to 0.56 section. Next time please split these changes, just so we can trace them back better. Or if it's trivial, please include all relevant tickets in the PR description. Anyone tracing back PRs should be able to find the reasoning behind changes.

### Remove raiseContainerWarning property

`raiseContainerWarning` property is removed from the following interfaces in release 0.56: `IContainerRuntime`, `IFluidDataStoreRuntime`, `IFluidDataStoreContext`, `IContainerRuntime`, `IFluidDataStoreRuntime`. It was deprecated in `IContainerContext` and it usage would be completely removed. The advised migration step for partners is to generate their own telemetry/logging events.

### Remove logger property from IContainerContext

> The `logger` property from `IContainerContext` is removed. The alternative is to use `taggedLogger` if possible. Loggers passed to `ContainerContext` will need to support tagged events.

Reviewer: ```suggestion
The `logger` property from `IContainerContext` is removed. Use `taggedLogger` instead. Loggers passed to `ContainerContext` will need to support tagged events.
```

+ import { IntervalType } from "@fluidframework/sequence";
```

## Remove logger property from IContainerContext

> The logger property in IContainerContext was set as an optional parameter in [release 0.56](#Set-logger-property-as-optional-parameter-in-IContainerContext) and have now been removed.

Reviewer 3: ```suggestion
The logger property in IContainerContext became an optional parameter in [release 0.56](#Set-logger-property-as-optional-parameter-in-IContainerContext). This property has now been removed.
```

The `id` property of IContainerContext was deprecated and now removed. The `id` property of ContainerContext was deprecated and now removed. id should not be exposed at
runtime level anymore. Instead, get from container's resolvedURL if necessary.

> ### Remove raiseContainerWarning property
> 
> `raiseContainerWarning` property is removed from the following interfaces in release 0.56: `IContainerRuntime`, `IFluidDataStoreRuntime`, `IFluidDataStoreContext`, `IContainerRuntime`, `IFluidDataStoreRuntime`. It was deprecated in `IContainerContext` and it usage would be completely removed. The advised migration step for partners is to generate their own telemetry/logging events.

Reviewer 3: ```suggestion
### Remove raiseContainerWarning property

The `raiseContainerWarning` property is removed from the following interfaces in release 0.56:

- `IContainerRuntime`
- `IFluidDataStoreContext`
- `IFluidDataStoreRuntime`

This property was also deprecated in `IContainerContext` and will be removed in a future release. Application developers should generate their own telemetry/logging events.
```

### Set logger property as optional parameter in IContainerContext

> The `logger` property from `IContainerContext` is set as an optional paramter and would be removed in the next release. Use `taggedLogger` instead. Loggers passed to `ContainerContext` will need to support tagged events.

Reviewer 3: ```suggestion
The `logger` property from `IContainerContext` is now optional. It will be removed completely in a future release. Use `taggedLogger` instead. Loggers passed to `ContainerContext` will need to support tagged events.
```
